public/app/plugins/datasource/prometheus/add_label_to_query.ts

```diff
@@ -35,7 +40,13 @@ export function addLabelToQuery(query: string, key: string, value: string, opera
     const isColonBounded = word.endsWith(':');
 
     previousWord = word;
-    if (!insideSelector && !isColonBounded && !previousWordIsKeyWord && builtInWords.indexOf(word) === -1) {
+    if (
```

Comment:
Could you add comment explaining why is this needed/what is happening here? That would be awesome!

Reply:
@ivanahuckova sure - it basically prevents from appending the filter for/out label to anything outside of the selector when using Loki datasource.

Right now, when you add a filter for/out a sentence with a dash character inside, e.g. `{selector} |= "foo-bar"`, adding a label from the table is going to add an empty selector to `foo` as well, and then fill it with the label - while it shouldn't.

Reply:
What about, instead of using `isLokiDatasource` here, we would use something like `hasNoMetrics` ? As `isLokiDatasource`. Also, it would remove Loki-specific code?

Reply:
@ivanahuckova good idea! Renamed it from `isLokiDatasource` to `hasNoMetrics`.